OpenRA.Mods.Common/Graphics/ContrailRenderable.cs

```diff
@@ -98,8 +100,16 @@ public void Render(WorldRenderer wr)
 
 				var nextPos = new WPos((int)(nextX / k), (int)(nextY / k), (int)(nextZ / k));
 
-				if (!world.FogObscures(curPos) && !world.FogObscures(nextPos))
-					wcr.DrawLine(wr.Screen3DPosition(curPos), wr.Screen3DPosition(nextPos), screenWidth, curColor, nextColor);
+				// renderLength can never be lower than 2. When it is 2 it's only one segment, so to avoid division by 0 we render startWidth instead
+				float width = startWidth.Length;
```

Comment:
```suggestion
				var width = startWidth.Length;
```

Reply:
this converts `int` to `float`

Reply:
Then it should be something like
```suggestion
				var width = (float)startWidth.Length;
```

Reply:
reformated